server/agents/explicit_query_process.py

```diff
@@ -39,11 +39,13 @@ def explicit_query_processing_loop():
                 if current_time > last_wake_word_time + force_query_time:
                     is_query_ready = True
 
-                # If there has been a pause
-                elif (len(user['final_transcripts']) != 0) and (current_time > user['final_transcripts'][-1]['timestamp'] + pause_query_time):
+                # If there has been a pause (the latest transcript is old)
+                latest_transcript = dbHandler.get_latest_transcript_from_user_obj(user)
+                if latest_transcript and (current_time > latest_transcript['timestamp'] + pause_query_time):
                     is_query_ready = True
 
                 if not is_query_ready: continue
+                
```

Comment:
line 85 below, where the insight gets generated, is it assumed that it would be like "Insight: this is an insight"? because in the other branch i changed insights to return a dict of { insight: str, resource_url: https://<link>, motive: str }

Reply:
ill test the code later, and will add some changes if it doesnt work out of the box